Scripts/Solo/AutoRebuild.as

```diff
@@ -0,0 +1,50 @@
+// Explanations are in AutoRebuild.cfg
+bool enabled;
+bool check_x_tick;
+s32 check_timer;
+bool check_focus;
+
+bool focused = false;
+
+void onInit(CRules@ this)
+{
+	onReload(this);
+}
+
+void onReload(CRules@ this)
+{
+	ConfigFile cfg = ConfigFile(CFileMatcher("AutoRebuild.cfg").getFirst());
+
+	enabled = cfg.read_bool("enabled", true);
+	check_x_tick = cfg.read_bool("check_x_tick", false);
+	check_timer = cfg.read_s32("check_timer", 90);
+	check_focus = cfg.read_bool("check_focus", true);
+
+	if (!enabled)
+	{
+		removeMe(this);
+	}    
+}
+
+void onTick(CRules@ this)
```

Comment:
Couldn't rebuild() be called twice on the same tick if both check_focus and check_x_tick are enabled? Could that be a problem?

Reply:
side from checking if we should rebuild twice this is not an issue